src/settings.rs

```diff
@@ -217,10 +217,17 @@ impl Settings {
     }
 
     #[must_use]
-    pub fn with_distinct_attribute(self, distinct_attribute: impl AsRef<str>) -> Settings {
-        Settings {
-            distinct_attribute: Some(distinct_attribute.as_ref().to_string()),
-            ..self
+    pub fn with_distinct_attribute(self, distinct_attribute: Option<impl AsRef<str>>) -> Settings {
+        if let Some(distinct_attribute) = distinct_attribute {
+            Settings {
+                distinct_attribute: Some(Some(distinct_attribute.as_ref().to_string())),
+                ..self
+            }
+        } else {
+            Settings {
+                distinct_attribute: Some(None),
+                ..self
+            }
```

Comment:
```suggestion
    pub fn with_distinct_attribute(self, distinct_attribute: Option<impl AsRef<str>>) -> Settings {
        Settings {
            distinct_attribute: Some(distinct_attribute.map(|distinct| distinct.to_string())),
            ..self
        }

```

Use the [`map`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.map) method on options